### PR TITLE
doc(quickstart): Sync HTTPUnauthorized example with post-0.3 changes

### LINKED ISSUE #757: More complex example crashes on any request
I tried to get the [more complex example](https://falcon.readthedocs.org/en/stable/user/quickstart.html#more-features) working and encountered the following error on any request:

```
2016-04-16 18:23:44 +0200] [13566] [ERROR] Error handling request /
Traceback (most recent call last):
  File "/Library/Python/2.7/site-packages/gunicorn/workers/sync.py", line 130, in handle
    self.handle_request(listener, req, client, addr)
  File "/Library/Python/2.7/site-packages/gunicorn/workers/sync.py", line 171, in handle_request
    respiter = self.wsgi(environ, resp.start_response)
  File "/Library/Python/2.7/site-packages/falcon/api.py", line 167, in __call__
    self._call_req_mw(middleware_stack, req, resp)
  File "/Library/Python/2.7/site-packages/falcon/api.py", line 535, in _call_req_mw
    process_request(req, resp)
  File "/Users/gigi/Development/newspaper3k-api-using-falcon/more-complex-example.py", line 61, in process_request
    href='http://docs.example.com/auth')
TypeError: __init__() takes exactly 4 arguments (3 given)
```

### PR DESCRIPTION
**Pull request by kgriffs**
Update the complex example given in the quickstart and README so that
it now works in light of a breaking change to the HTTPUnauthorized
class. Note this breaking change in the changelog.

Closes #757

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -152,7 +152,7 @@ API.
 
     # things.py
 
-    # Let's get this party started
+    # Let's get this party started!
```
Comment (fxfitz) [inline README.rst]: I'm so +1 to this

File: doc/user/quickstart.rst [hunk 2]
```diff
@@ -207,7 +210,7 @@ parameters, handling errors, and working with request and response bodies.
         return hook
 
 
-    class ThingsResource(object):
+    class ThingsResource:
```
Comment (fxfitz) [inline doc/user/quickstart.rst]: Why are we switching from new style classes to old style??

**Comment by fxfitz** [comment]
LGTM, just want to double check about the new->old-style class change.

File: doc/user/quickstart.rst (hunk 2 above)
Comment (kgriffs) [inline doc/user/quickstart.rst]: Hmm, not sure how that slipped in there. I'll fix it pronto!

Comment (kgriffs) [inline doc/user/quickstart.rst]: Fixed. Nice catch!

**Comment by fxfitz** [comment]
LGTM

**Pull request merged**